fili-core/src/test/groovy/com/yahoo/bard/webservice/async/AsyncDruidSendsErrorSpec.groovy

```diff
@@ -0,0 +1,90 @@
+
+import static com.yahoo.bard.webservice.async.jobs.jobrows.DefaultJobStatus.FAILURE
+import static com.yahoo.bard.webservice.async.jobs.jobrows.DefaultJobStatus.PENDING
+
+import com.yahoo.bard.webservice.util.GroovyTestUtils
+
+/**
+ * Verifies that the job status is updated to error, and results links return the error message when Druid experiences
+ * an error.
+ */
+class AsyncDruidSendsErrorSpec extends AsyncFunctionalSpec {
+
+    static final String QUERY =
+            "http://localhost:9998/data/shapes/day?dateTime=2016-08-30%2F2016-08-31&metrics=height&asyncAfter=0"
+
+    static final String ERROR_MESSAGE = """{
+                        "status" : 500,
+                        "statusName" : "Internal Server Error",
+                        "reason" : "All the things have broken.",
+                        "description" : "All the things have broken.",
+                        "druidQuery" : null
+                    }"""
+
+    @Override
+    Map<String, Closure<String>> getResultsToTargetFunctions() {
+        return [
+                data: { "data/shapes/day" },
+                //By querying the syncResults link first, we wait until the results are ready, thanks to the
+                //BroadcastChannel. So we have no race condition between the data request updating the job ticket, and
+                //the query for the job metadata
+                syncResults: { AsyncTestUtils.extractTargetFromField(it.data.readEntity(String), "syncResults") },
+                results: { AsyncTestUtils.extractTargetFromField(it.data.readEntity(String), "results") },
+                jobs: { AsyncTestUtils.buildTicketLookup(it.data.readEntity(String)) }
+        ]
+    }
+
+    @Override
+    Map<String, Closure<Void>> getResultAssertions() {
+        return [
+                data: {
+                    assert it.status == 202
+                    AsyncTestUtils.validateJobPayload(it.readEntity(String), QUERY, PENDING.name)
+                },
+                syncResults: {
+                    assert it.status == 500
+                    assert GroovyTestUtils.compareJson(it.readEntity(String), ERROR_MESSAGE)
+                },
+                results: {
+                    assert it.status == 500
+                    assert GroovyTestUtils.compareJson(it.readEntity(String), ERROR_MESSAGE)
+                },
+                jobs: { response ->
+                    assert response.status == 200
+                    AsyncTestUtils.validateJobPayload(response.readEntity(String), QUERY, FAILURE.name)
+                }
+        ]
+    }
+
+    @Override
+    Map<String, Closure<Map<String, List<String>>>> getQueryParameters() {
+        return [
+                data: {[
+                        metrics: ["height"],
+                        asyncAfter: ["0"],
+                        dateTime: ["2016-08-30/2016-08-31"]
+                ]},
+                jobs: {[:]},
+                results: {[:]},
+                syncResults: {
```

Comment:
The ordering of this map should (for clarity at least) match the ordering of the other closure maps.